crates/sc-consensus-subspace/src/lib.rs

```diff
@@ -1075,6 +1118,43 @@ where
 
         Ok(())
     }
+
+    /// Verifies the proof of time in the received block.
+    #[allow(clippy::too_many_arguments)]
+    fn proof_of_time_verification(
+        &self,
+        proof_of_time: &dyn PotConsensusState,
+        block_number: NumberFor<Block>,
+        pre_digest: &PreDigest<FarmerPublicKey, FarmerPublicKey>,
+        parent_pre_digest: &PreDigest<FarmerPublicKey, FarmerPublicKey>,
+    ) -> Result<(), PotVerifyError<Block>> {
+        let parent_pot_digest = parent_pre_digest.proof_of_time.as_ref().ok_or_else(|| {
+            PotVerifyError::ParentMissingPotDigest {
+                block_number,
+                parent_slot_number: parent_pre_digest.slot.into(),
+                slot_number: pre_digest.slot.into(),
+            }
```

Comment:
As we discussed, once we have the global randomness from PoT in next change, the slot worker will block until the local chain catches up. Then we won't be producing blocks without PoT in them

Reply:
I mean my assumption was that you can test something with this PR, but according to this check you wouldn't be able to produce second block (wouldn't be able to import it to be more specific). Am I right?

Reply:
Sorry I take this back. claim_slot() currently errors out if proof creation fails, so we won't be creating blocks with no PoT proof. But this will be changed to block if the chain is behind